wp-modules/editor/tests/ModelTest.php

```diff
@@ -198,4 +198,89 @@ public function test_slug_and_filename_stay_the_same_after_content_update() {
 		// Make sure the filename does not get changed when only the content is modified.
 		$this->assertSame( 'mismatched-name', $content_modified_pattern['name'] );
 	}
+	
+	/**
+	 * Tests that images remain after a pattern is renamed.
+	 */
+	public function test_images_remain_after_a_pattern_is_renamed() {
+		
+		$wp_filesystem = get_wp_filesystem_api();
+		
+		// Save the post using the REST api, which triggers saving the file.
+		wp_set_current_user(1);
+			
+		$content = '<!-- wp:image {"id":610,"sizeSlug":"full","linkDestination":"none"} -->
+			<figure class="wp-block-image size-full"><img src="https://upload.wikimedia.org/wikipedia/commons/thumb/6/6e/Golde33443.jpg/220px-Golde33443.jpg" alt="" class="wp-image-610"/></figure><!-- /wp:image -->';
+	
+		$request = new \WP_REST_Request( 'POST', '/wp/v2/posts' );
```

Comment:
This won't run the update_post_metadata [callback](https://github.com/studiopress/pattern-manager/pull/192/commits/4ce23ee7d8dafbc5b507b7825ab904082c482898#diff-8ba31aa77fc662d19b7f20a487d3c2e0d83ac9ab99b843479f7ddae96b0649edR166), it's separate from it.

The best we could do is simulate the hooks that run in the block editor:

```
apply_filters( 'get_post_metadata',  $mock_override, $mock_post_id, $mock_meta_key, $mock_meta_value  );
do_action( 'rest_after_insert_' . get_pattern_post_type(), $mock_post );
```

…though maybe it's better to call directly `save_metadata_to_pattern_file()` and `save_pattern_to_file()`.